CMakeLists.txt

```diff
@@ -126,7 +116,7 @@ if (TARGET gridtools_cuda)
     target_link_libraries(storage_cuda INTERFACE gridtools gridtools_cuda)
     list(APPEND GT_STORAGES cuda)
 
-    if(TARGET MPI::MPI_CXX)
+    if(MPI_CXX_FOUND)
```

Comment:
Is there the difference between `if(MPI_CXX_FOUND)` and  `if(TARGET MPI::MPI_CXX)` ? If not I would prefer the `TARGET` form to minimize the set of used global variables.

Reply:
This is probably a bug in the FindMPI: the target is set even if detection fails.